### ISSUE TITLE
Different on total point counts

### ISSUE THREAD

**Issue by lapd-viosoft** [carla-simulator/carla#4031]
Hi,
I'm using C++ example with Carla 0.9.11 to get total points from lidar and I found that, total point count are not the same on the way we count it.
```c
    // Register a callback to receive data
    lidar->Listen([](auto data) {
      auto lidar_data = boost::static_pointer_cast<csd::LidarMeasurement>(data);
      assert(lidar_data != nullptr);

      /////////////////////////////////////////////////////////////////////////
      ///////////////////////// FIRST METHOD //////////////////////////////////
      /////////////////////////////////////////////////////////////////////////
      int totalPoints1 = 0;
      for(unsigned int j = 0; j < lidar_data->GetChannelCount() ; j++){
        totalPoints1 += lidar_data->GetPointCount(j);
      }
      std::cout << "totalPoints1=" << totalPoints1  << std::endl;
      /////////////////////////////////////////////////////////////////////////
      ///////////////////////// SECOND METHOD //////////////////////////////////
      /////////////////////////////////////////////////////////////////////////    
      int totalPoints2 = 0;
      auto iter = lidar_data->begin();
      while (iter != lidar_data->end()){
          totalPoints2++;
          iter++;
      }
      std::cout << "totalPoints2=" << totalPoints2  << std::endl;

      // Just for debug
      exit(0);
    });
````

```sh
Client API version : 0.9.11
Server API version : 0.9.11
totalPoints1=12232
totalPoints2=12221
```

Can you please tell me, which one is correct way to get total points?
Thank you!

**Comment by corkyw10** [comment]
@DSantosO could you follow up on this please?

**Comment by DSantosO** [comment]
Hello lapd-viosoft,
Thank you for your report.
We have reproduced the issue and we are looking into it.

**Comment by DSantosO** [comment]
Hello @lapd-viosoft,
There was a bug in the computation of the channel point count. It is fixed now in PR #4075 and it will be available in `dev` branch soon.
Thank you for your report.
